wagtail_extensions/blocks.py

```diff
@@ -24,17 +26,15 @@ def get_prep_value(self, value):
 
 
 class LinkBlock(blocks.StructBlock):
-    text = blocks.CharBlock()
+    text = blocks.CharBlock(required=False)
     page = blocks.PageChooserBlock(required=False)
     absolute_url = blocks.URLBlock(label="Url", required=False)
 
     class Meta:
         template = 'wagtail_extensions/blocks/link.html'
 
     def clean(self, value):
-        page = value.get('page')
-        absolute_url = value.get('absolute_url')
-        if (not page and not absolute_url) or (page and absolute_url):
+        if value.get('page') and value.get('absolute_url'):
```

Comment:
This checks that both are not set, but you've now lost the check that makes sure that one of them *is* set.